File: benchmark/corpus/04_off_by_one/c_skips_last.py
# Off by one the other way: the loop stops early and the total silently omits the last item. No read is out of range and no contract exists, so nothing can flag it.


def total(xs):
    s = 0
    for i in range(0, len(xs) - 1):  # DANGER
        s = s + xs[i]
    return s


print(total([2500, 45000, 12000]))

File: benchmark/corpus/04_off_by_one/c_skips_last.js
// Off by one the other way: the loop stops early and the total silently omits the last item. No read is out of range and no contract exists, so nothing can flag it.
function total(xs) {
  let sum = 0;
  for (let i = 0; i < xs.length - 1; i++) { // DANGER
    sum = sum + xs[i];
  }
  return sum;
}

console.log(total([2500, 45000, 12000]));
